### PR TITLE
Feat/admin api rebase v2

### PR DESCRIPTION
**Pull request by pj-cegeka**

### CODE REVIEW

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/exceptions/MissingLdesConfigException.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	/**
+	 * Constructs a new MissingLdesConfigException with default object type <i>ldes
+	 * event stream</i>
+	 *
+	 * @param identifier
+	 *            identifier of the missing ldes config object
+	 */
+	public MissingLdesConfigException(String identifier) {
+		super();
+		this.objectType = "ldes event stream";
```
Comment (Yalz) [inline ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/exceptions/MissingLdesConfigException.java:22]: LDES Event Stream is a hat on a hat ;)

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/repository/LdesConfigRepository.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesconfig.repository;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesconfig.valueobjects.LdesConfigModel;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface LdesConfigRepository {
+	List<LdesConfigModel> retrieveAllLdesStreams();
```
Comment (sverholen) [inline ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/repository/LdesConfigRepository.java:9]: xxxConfig(s)

Comment (sverholen) [reply ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/repository/LdesConfigRepository.java:9]: ___ConfigModel(s)

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/services/LdesConfigModelService.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+	LdesConfigModel retrieveEventStream(String collectionName);
+
+	void deleteEventStream(String collectionName);
+
+	LdesConfigModel updateEventStream(LdesConfigModel ldesConfigModel);
+
+	LdesConfigModel retrieveShape(String collectionName);
+
+	LdesConfigModel updateShape(String collectionName, LdesConfigModel shape);
```
Comment (sverholen) [inline ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/services/LdesConfigModelService.java:19]: ___ConfigModel(s)

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/services/LdesConfigModelServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+		Model shapeModel = ldesConfigModel.getModel().listStatements(null,
+				createProperty(SHAPE), (Resource) null).toList().stream().findFirst()
+				.map(statement -> retrieveAllStatements(statement.getResource(), ldesConfigModel.getModel()))
+				.map(statements -> {
+					Model model = ModelFactory.createDefaultModel();
+					model.add(statements);
+					return model;
+				})
+				.orElse(ModelFactory.createDefaultModel());
+		return new LdesConfigModel(collectionName + "Shape", shapeModel);
```
Comment (sverholen) [inline ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/services/LdesConfigModelServiceImpl.java:63]: move to a "factory method" pls

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/services/LdesConfigModelServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+	@Override
+	public void deleteView(String collectionName, String viewName) {
+		LdesConfigModel ldesConfigModel = retrieveEventStream(collectionName);
+
+		StmtIterator iterator = ldesConfigModel.getModel().listStatements(idStringToResource(collectionName),
+				createProperty(VIEW), idStringToResource(viewName));
+
+		if (!iterator.hasNext()) {
+			throw new MissingLdesConfigException("view", collectionName + "/" + viewName);
```
Comment (sverholen) [inline ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldesconfig/services/LdesConfigModelServiceImpl.java:132]: Move this to an enum

**Review by sverholen — APPROVED** [review]

**Pull request merged**